app/code/Magento/InventoryInStorePickupQuote/Plugin/Quote/Address/LoadPickupLocationForQuoteAddress.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupQuote\Plugin\Quote\Address;
+
+use Magento\InventoryInStorePickupQuote\Model\ResourceModel\GetPickupLocationCodeByQuoteAddressId;
+use Magento\Quote\Api\Data\AddressExtensionInterfaceFactory;
+use Magento\Quote\Api\Data\AddressInterface;
+use Magento\Quote\Model\ResourceModel\Quote\Address;
+
+/**
+ * Load Pickup Location Code for Quote Address.
+ */
+class LoadPickupLocationForQuoteAddress
+{
+    /**
+     * @var GetPickupLocationCodeByQuoteAddressId
+     */
+    private $getPickupLocationCodeByQuoteAddressId;
+
+    /**
+     * @var AddressExtensionInterfaceFactory
+     */
+    private $addressExtensionInterfaceFactory;
+
+    /**
+     * LoadPickupLocationForQuoteAddress constructor.
+     *
+     * @param GetPickupLocationCodeByQuoteAddressId $getPickupLocationCodeByQuoteAddressId
+     * @param AddressExtensionInterfaceFactory $addressExtensionInterfaceFactory
+     */
+    public function __construct(
+        GetPickupLocationCodeByQuoteAddressId $getPickupLocationCodeByQuoteAddressId,
+        AddressExtensionInterfaceFactory $addressExtensionInterfaceFactory
+    ) {
+        $this->getPickupLocationCodeByQuoteAddressId = $getPickupLocationCodeByQuoteAddressId;
+        $this->addressExtensionInterfaceFactory = $addressExtensionInterfaceFactory;
+    }
+
+    /**
+     * @param Address $subject
+     * @param Address $result
+     * @param AddressInterface $entity
+     *
+     * @return Address
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterLoad(Address $subject, Address $result, AddressInterface $entity): Address
+    {
+        if (!$entity->getId()) {
+            return $result;
+        }
+
+        $pickupLocationCode = $this->getPickupLocationCodeByQuoteAddressId->execute((int)$entity->getId());
+        if ($pickupLocationCode) {
```

Comment:
Please process the negative case first to avoid nested **if** operators and reduce code complexity.

Reply:
Done.